report/5.Solution/DockerBuild.tex

```diff
@@ -0,0 +1,113 @@
+\section{Build process}\label{sec:buildprocess}
+In previous semesters, the build process has been straightforward, as it has simply been a matter of using the tool included with the IDE of choice for the chosen language.
+This semester the build process is quite a bit more complicated as the project targets the NXT.
+This complicates the build process for a number of reasons:
+nxtOSEK is the operating system for RTS on the NXT, and even though it is quite old, the latest release, as of the writing of this report, was released in January of 2013\cite{osekrelease}.
+
+During the groups testing of the nxtOSEK build, it was found that it was largely incompatible with modern operating systems, both Windows, OS X and a number of modern Linux distributions.
+According to the latest documentation on the nxtOSEK webpage, the latest supported version of Linux is Ubuntu 11.10 which was released in 2011.
+This complicated things a bit, as none of the group members were interested in installing such an old operating system on their personal computer, and instead Docker was used to avoid this.
+
+\subsection{Docker}\label{subsec:docker}
+Docker is a way to deploy a consistent environment for applications to run in, also known as containerization\cite{dockerdocstart}.
+A Docker deployment consists of three different concepts, the Docker file, the container and an image.
+\begin{description}
+    \item [Dockerfile], which is a definition of a Docker image.
+    \item [Images] are executables that includes all the required information to instantiate containers.
+    \item[Containers] are the runtime instances of an image, the running process of an image.
+\end{description}
+
+A Dockerfile simply consists of lines, that describe how to build the image.
+Each line applies a layer to the Docker image, and each of these layers are cached.
+This means, once the Docker image is built, these will not be re-executed.
```

Comment:
"not changed - unless changed". Hvis man ændrer fx linje 1, bliver ALT andet rebuilt også, men hvis du ændrer linje 5 er det 5+ der rebuildes osv.